bika/lims/idserver.py

```diff
@@ -49,148 +49,103 @@ def idserver_generate_id(context, prefix, batch_size = None):
-            separator = e['separator']
-
-    # Analysis Request IDs
-    if context.portal_type == "AnalysisRequest":
-        sample = context.getSample()
-        s_prefix = fn_normalize(sample.getSampleType().getPrefix())
-        sample_padding = context.bika_setup.getSampleIDPadding()
-        ar_padding = context.bika_setup.getARIDPadding()
-        sample_id = sample.getId()
-        sample_number = sample_id.split(s_prefix)[1]
-        ar_number = sample.getLastARNumber()
-        ar_number = ar_number and ar_number + 1 or 1
-        return fn_normalize(
-            ("%s%s" + separator + "R%s") % (s_prefix,
-                          str(sample_number).zfill(sample_padding),
-                          str(ar_number).zfill(ar_padding))
-        )
-
-    # Sample Partition IDs
-    if context.portal_type == "SamplePartition":
-        # We do not use prefixes.  There are actually codes that require the 'P'.
-        # matches = [p for p in prefixes if p['portal_type'] == 'SamplePartition']
-        # prefix = matches and matches[0]['prefix'] or 'samplepartition'
-        # padding = int(matches and matches[0]['padding'] or '0')
-
-        # at this time the part exists, so +1 would be 1 too many
-        partnr = str(len(context.aq_parent.objectValues('SamplePartition')))
-        # parent id is normalized already
-        return ("%s" + separator + "P%s") % (context.aq_parent.id, partnr)
-
-    if context.bika_setup.getExternalIDServer():
-
-        # if using external server
-
-        for d in prefixes:
-            # Sample ID comes from SampleType
-            if context.portal_type == "Sample":
-                prefix = context.getSampleType().getPrefix()
-                padding = context.bika_setup.getSampleIDPadding()
-                new_id = str(idserver_generate_id(context, "%s%s-" % (prefix, year)))
-                if padding:
-                    new_id = new_id.zfill(int(padding))
-                return ('%s%s' + separator + '%s') % (prefix, year, new_id)
-            elif d['portal_type'] == context.portal_type:
-                prefix = d['prefix']
-                padding = d['padding']
-                new_id = str(idserver_generate_id(context, "%s%s-" % (prefix, year)))
-                if padding:
-                    new_id = new_id.zfill(int(padding))
-                return ('%s%s' + separator + '%s') % (prefix, year, new_id)
-        # no prefix; use portal_type
-        # year is not inserted here
-        # portal_type is be normalized to lowercase
-        npt = id_normalize(context.portal_type)
-        new_id = str(idserver_generate_id(context, npt + "-"))
-        return ('%s' + separator + '%s') % (npt, new_id)
+    if portal_type == '':
+        portal_type = context.portal_type
 
+    def getLastCounter(context, config):
+        if config.get('counter_type', '') == 'backreference':
+            return len(context.getBackReferences(config['counter_reference'])) - 1
```

Comment:
@rockfruit: Will this will work with the new `UIDReference` field?

Reply:
Nope.  I will implement an attribute based version of backreferences for UIDReferenceField now.